joy/src/joy_node_linux.cpp

```diff
@@ -240,7 +243,10 @@ class Joystick
             break; // Joystick is probably closed. Definitely occurs.
           
           //ROS_INFO("Read data...");
-          joy_msg->header.stamp = rclcpp::Time::now();
+          rclcpp::TimeSource ts(node);
+          rclcpp::Clock::SharedPtr clock = std::make_shared<rclcpp::Clock>(RCL_ROS_TIME);
+          ts.attachClock(clock);
```

Comment:
Can this be moved outside of the while loop rather than creating these objects every time we send a message?

Reply:
Good call on that, done in 263bf54